ar_exercise/exercise.rb

```diff
@@ -32,40 +32,92 @@ class Apartment < ActiveRecord::Base
+#puts damien_corners.address
+
 # get all tenants in that apartment
+damien_corners_tenants = Apartment.find {|ap| ap.address = "6005 Damien Corners"}.tenants
+#damien_corners_tenants.each{|t| puts t.name}
 
 # Use `each` and `puts` to:
 # Display the name and ID # of every tenant
+all_tenants.each {|t| puts "Tenant ID: #{t.id}; Tenant Name:'#{t.name}';"}
 # Iterate over each apartment, for each apartment, display it's address and rent price
+Apartment.all.each {|ap| puts "Apartment ID: #{ap.id}; Apartment Address: #{ap.address}; Apartment Rent: #{ap.monthly_rent}"}
 # Iterate over each apartment, for each apartment, display it's address and all of it's tenants
+Apartment.all.each do |ap|
+  puts "Apartment ID: #{ap.id};"
+  ap.tenants.each {|t| puts t.name}
+end
 
 ################################################
 # CREATING / UPDATING / DELETING
 ################################################
 
-# Hint, the following methods will help: `new`, `create`, `save`, `uddate`, `destroy`
+# Hint, the following methods will help: `new`, `create`, `save`, `update`, `destroy`
 
 # Create 3 new apartments, and save them to the DB
+
+Apartment.where('address=? OR address=? OR address=?',"International Space Station #1", "Giza Pyramid", "Alexanria Library").destroy_all
+
+apartment1= Apartment.create(address:"International Space Station #1", monthly_rent: 999999, sqft: 400, num_beds: 0, num_baths: 0)
+apartment2= Apartment.create(address:"Giza Pyramid", monthly_rent: 99999, sqft: 50000, num_beds: 10, num_baths: 5)
+apartment3= Apartment.create(address:"Alexanria Library", monthly_rent: 99999, sqft: 10000, num_beds: 10, num_baths: 5)
+#Apartment.all.each {|ap| puts "Apartment ID: #{ap.id}; Apartment Address: #{ap.address}; Apartment Rent: #{ap.monthly_rent}"}
+
+puts "*"*30
 # Create at least 9 new tenants and save them to the DB. (Make sure they belong to an apartment)
 # Note: you'll use this little bit of code as a `seeds.rb` file later on.
-
+# Tenant.where('age > ?', 100).each {|t| puts t.name} # how to convert Age to integer
+Tenant.where('age > ?', 100).destroy_all
+
+superBestFriends1 = Tenant.create(name:"Jesus" , age:2016 , gender: 'male', apartment_id:5 )
+superBestFriends2 = Tenant.create(name:"Muhammad" , age:1616 , gender:'male' , apartment_id:5 )
+superBestFriends3 = Tenant.create(name:"Buddha" , age:4000 , gender:'male' , apartment_id:5 )
+superBestFriends4 = Tenant.create(name:"Moses" , age:1900 , gender:'male' , apartment_id:5 )
+superBestFriends5 = Tenant.create(name:"Joseph Smith" , age: 452 , gender:'male' , apartment_id:5 )
+superBestFriends6 = Tenant.create(name:"Krishna" , age:6000 , gender:'male' , apartment_id:5 )
+superBestFriends7 = Tenant.create(name:"Laozi" , age:3500 , gender:'male' , apartment_id:5 )
+superBestFriends8 = Tenant.create(name:"Sea Man" , age:1000 , gender:'male' , apartment_id:5 )
+superBestFriends9 = Tenant.create(name:"Chuhutlu" , age:1000 , gender:'male' , apartment_id:5 )
+
```

Comment:
You've hard coded the apartment_id, which generally a bad practice. Fortunately ActiveRecord is super smart and will add the appropriate apartment_id when we create a tenant on an apartment
```rb
apartment1.tenants.create("a bunch of key value pairs, not including apartment_id")
```